clients/cli/src/create_mint.rs

```diff
@@ -120,7 +123,17 @@ pub async fn command_create_mint(config: &Config, args: CreateMintArgs) -> Comma
         Err(_) => 0,
     };
 
-    let mint_size = DefaultToken2022Customizer::get_token_2022_mint_space()?;
+    let mint_size = if args.wrapped_token_program == spl_token_2022::id() {
+        let base_size = DefaultToken2022Customizer::get_token_2022_mint_space()?;
+        // The TokenMetadata extension is initialized *after* and its
+        // `initialize` instruction handles its own reallocation.
+        let metadata_size = TokenMetadata::default().tlv_size_of()?;
+        base_size
+            .checked_add(metadata_size)
+            .ok_or(ProgramError::ArithmeticOverflow)?
+    } else {
+        spl_token::state::Mint::LEN
+    };
```

Comment:
Kinda sad we can't straight use the customizer method as space calculation is quite weird given the later realloc.